helfi_features/helfi_events/src/EventsApiBase.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace Drupal\helfi_events;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\helfi_events\Enum\CategoryKeywords;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Base class for retrieving events data.
+ */
+class EventsApiBase {
+  /**
+   * The cache.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected CacheBackendInterface $dataCache;
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * The constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $httpClient
+   *   HTTP Client.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $dataCache
+   *   Data Cache.
+   */
+  public function __construct(ClientInterface $httpClient, CacheBackendInterface $dataCache) {
```

Comment:
These can be replaced to use constructor parameter promotion since we're using 8.0+ only: https://stitcher.io/blog/constructor-promotion-in-php-8